src/Infrastructure/BotSharp.Abstraction/Knowledges/IPdf2TextConverter.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNetCore.Http;
+
+namespace BotSharp.Abstraction.Knowledges
+{
+    public interface IPdf2TextConverter
+    {
+        Task<string> ConvertPdfToText(IFormFile formFile, int? startPageNum, int? endPageNum);
+        Task<string> OpenPdfDocumentAsync(IFormFile formFile, int? startPageNum, int? endPageNum);
```

Comment:
I think these 4 functions don't need to be exposed outside. They're used internally. So we can remove these interfaces.
- OpenPdfDocumentAsync
- LocalImageToTextsAsync
- ConvertPdfToLocalImagesAsync
- DeleteTempFile